jsonschema-maven-plugin/src/main/java/com/github/victools/jsonschema/plugin/maven/SchemaGeneratorMojo.java

```diff
@@ -187,21 +186,13 @@ private void generateSchema(Class<?> schemaClass) throws MojoExecutionException
      * @throws MojoExecutionException in case of problems
      */
     private void generateSchemaForPackage(String packageName) throws MojoExecutionException {
-        Reflections reflections = new Reflections(this.getClassLoader(), packageName, new SubTypesScanner(false));
-        Set<Class<?>> subTypes;
-        try {
-            subTypes = reflections.getSubTypesOf(Object.class);
-        } catch (ReflectionsException reflectionsException) {
-            if (reflectionsException.getMessage().contains("Scanner SubTypesScanner was not configured")) {
-                throw new MojoExecutionException("Could not locate any classes in package " + packageName, reflectionsException);
-            } else {
-                throw new MojoExecutionException("Unknown problem when locating classes in package " + packageName, reflectionsException);
+        Reflections reflections = new Reflections("", new SubTypesScanner(false), this.getClassLoader());
+        Set<String> classes = reflections.getAllTypes();
+        for (String className : classes) {
+            if (className.startsWith(packageName)) {
```

Comment:
I suspect false positives with this, when you have a `<packageName>com.my.app</packageName>` also picking up types from neighbouring packages like `com.my.appservice.Neighbour`.

Seeing this complete collection of all types here, I'm wondering whether this would be a good opportunity to address GH-104 by introducing some kind of glob matching.
For that, having the concept of including and excluding certain patterns would be helpful instead of the strict separation into `<className>` and `<packageName>`. The main challenge I see here would be backward-compatibility, but I guess we could keep them side-by-side for a while.